### PR TITLE
Angular 1.5

### PR DESCRIPTION
**Pull request by marcochavezf**
Controller and Route changed to use .component() from angular 1.5, I test it with another PR I did to angular-fullstack: https://github.com/angular-fullstack/generator-angular-fullstack/pull/1650

Have a good weekend!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
+.idea/
```
Comment (Awk34) [inline .gitignore]: this has nothing to do with either of your commits

File: package.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
   ],
   "dependencies": {
     "chalk": "~0.4.0",
-    "lodash": "^3.10.1",
+    "lodash": "^4.5.0",
```
Comment (Awk34) [inline package.json]: this has nothing to do with either of your commits

Comment (marcochavezf) [inline package.json]: I update it in order to use lodash.lowerFirst() in script-base.js, because that function is only available at 4.5.0, but I could use another method to lowercase the first char :)

Comment (Awk34) [inline package.json]: Bumping lodash is fine, as long as you check the breaking changes and do it in a separate commit. We want to make sure we keep separate concerns. (`chore(npm): lodash ^4.5.0`, then the rest)

File: templates/controller/name.controller.js [hunk 3]
```diff
@@ -1,6 +1,19 @@
 'use strict';
```
Comment (Awk34) [inline templates/controller/name.controller.js]: I don't believe this should be changed. The component type is pretty much just a specific type of directive shorthand. The controller subgenerator *just* creates a controller and has nothing to do with directives.

**Comment by Awk34** [comment]
So changing the route subgenerator to use a component is fine, but the controller subgenerator shouldn't use a component since it's **only** for making a controller, not a directive. Also if you feel up for it, #30 is for creating a new component subgenerator.

Another thing is naming conventions. If I give the subgenerator `myName`/`my name`/etc, the controller class should be named `MyNameComponent` (_.camelCase + _.upperFirst), and the custom element name should be `my-name` (kebab case).

**Comment by marcochavezf** [comment]
I removed the controller subgenerator changes and updated the naming convention for the route subgenerator.

**Comment by Awk34** [comment]
I have fixed the commits and put into #33

**Comment by marcochavezf** [comment]
Thanks @Awk34, I'm going to be more careful in the next contributions. 😄
